loadgen/adapters/orderer.go

```diff
@@ -83,25 +92,28 @@ func (*OrdererAdapter) Supports() Phases {
 	}
 }
 
-// sendTransactions submits Fabric TXs. It uses the envelope's TX ID to track the TXs latency.
-func (c *OrdererAdapter) sendTransactions(
-	ctx context.Context, txStream *workload.StreamWithSetup, stream *broadcastdeliver.EnvelopedStream,
-) error {
-	txGen := txStream.MakeTxGenerator()
-	for ctx.Err() == nil {
-		tx := txGen.Next(ctx, 1)
-		if len(tx) == 0 {
-			// The context ended.
-			return nil
-		}
-		txID, err := stream.SendWithEnv(tx[0])
+// mapToBatch creates a batch of orderer envelopes. It uses the envelope header ID to track the TXs latency.
+func mapToBatch(
+	txs []*protoblocktx.Tx, stream *broadcastdeliver.EnvelopedStream,
+) ([]*common.Envelope, []string, error) {
+	envs := make([]*common.Envelope, len(txs))
+	txIDs := make([]string, len(txs))
+	for i, tx := range txs {
+		var err error
+		envs[i], txIDs[i], err = stream.CreateEnvelope(tx)
```

Comment:
We have an issue to parallelize this. This change may make it hard to achieve that.